### DISCUSSION TITLE
Next JS: Server Side Rendering ~ Params Undefined

### DISCUSSION THREAD

**Discussion by abdullahmujahidali** [vercel/next.js#29471] [Help] [answered]
am working on an SSR (Next JS application). My route is http://127.0.0.1:3000/[pid] Example route is http://127.0.0.1:3000/test-1

In my getServerSideProps, I am printing params and it is returning undefined. I don't know why. I should print 'test-1'

IGNORE THE REST OF THE FUNCTION MY MAIN CONCERN IS PARAMS HERE

THANKYOU

```
export async function getServerSideProps (context) {
const { params } = context;
console.log("PARAMS: ", params)

const pid= 'test-1'
const result = await api.getTenantInfo('test-1');
const data = await result.data;

return {
props: {
      tenants: data,
      pid
},
  }
 }
```
The data which I am hardcoding in the APIs are supposed to be retrieved by context

LINK TO STACKOVERFLOW: https://stackoverflow.com/questions/69366509/next-js-server-side-rendering-params-undefined

**Comment by ijjk** [comment]
Hi, can you provide a link to a repo with a minimal reproduction? Do you have any rewrites configured in your `next.config.js`, what version of Next.js are you using?

**Reply by abdullahmujahidali** [reply]
> Hi, can you provide a link to a repo with a minimal reproduction? Do you have any rewrites configured in your `next.config.js`, what version of Next.js are you using?

Actually the repo is private however the next.config.js file is below:
`const withImages = require('next-images')
module.exports = withImages()

module.exports = withImages({
    images: {
        loader: "imgix",
        path: "",
    },
    exportPathMap: async function (
        defaultPathMap,
        { dev, dir, outDir, distDir, buildId }
      ) {
        return {
          '/': { page: '/' },
          '/:tenantUname': { page: '/[pid]' },
          '/:tenanUname/:jobId': {page: '/[pid]/[jobid]'}
        }
      },
  })`

**Comment by abdullahmujahidali** [comment, marked as answer]
So the problem was with my next.config.js file. As I was doing Server Side Rendering so I don't have to provide exportPathMap to my configuration file. After removing that it's working fine.

 ```
   const withImages = require('next-images')
    module.exports = withImages()

    module.exports = withImages({
      images: {
        loader: "imgix",
        path: "",
       },
    })

```
Above is my nest.config.js file for reference to someone who was facing the same problem.
